cmd/main.go

```diff
@@ -259,18 +283,32 @@ func main() {
 }
 
 // run starts the controllers and blocks until they stop.
-func run(ctx context.Context, kubeClient kubernetes.Interface, er record.EventRecorder, edgelbManager manager.EdgeLBManager) {
-	// Create a shared informer factory for the base API types.
-	kubeInformerFactory := kubeinformers.NewSharedInformerFactory(kubeClient, resyncPeriod)
-	// Create a cache for Kubernetes resources based on the shared informer factory.
-	kubeCache := cache.NewInformerBackedResourceCache(kubeInformerFactory)
+func run(ctx context.Context, kubeClient kubernetes.Interface, er record.EventRecorder, edgelbManager manager.EdgeLBManager, kubeInformerFactory kubeinformers.SharedInformerFactory, kubeCache dklbcache.KubernetesResourceCache, dcosClient *dcos.APIClient, saConfig dcos.ServiceAccountOptions) {
```

Comment:
nit: Too may parameters. Could be refactored to take a struct with fields set - easier for refactoring/extending in future.

Not a blocker though, especially when it's used only in this one place.

Reply:
I know and I have separate branch with a refactor going on. I'll add this to that one.